test/test_testing.py

```diff
@@ -1147,5 +1147,69 @@ def test_mismatching_device_no_check(self, devices):
 instantiate_device_type_tests(TestAssertsMultiDevice, globals(), only_for="cuda")
 
 
+class TestAssertsSparse(TestCase):
+    def test_matching(self):
+        indices = (
+            (0, 1),
+            (1, 0),
+        )
+        values = (1, 2)
+        actual = torch.sparse_coo_tensor(indices, values, (2, 2))
+        expected = actual.clone()
+
+        for fn in assert_fns_with_inputs(actual, expected):
+            fn()
+
+    def test_matching_uncoalesced(self):
+        indices = (
+            (0, 1, 0,),
+            (1, 0, 1,),
+        )
+        values = (1, 1, 1)
+        actual = torch.sparse_coo_tensor(indices, values, (2, 2))
+        expected = actual.clone().coalesce()
+
```

Comment:
Both `actual` and `expected` are coalesced, hence nothing related to uncoalesced state is tested as the test name is indicating. I suggest changing this to
```
actual = torch.sparse_coo_tensor(indices, values, (2, 2))
expected = actual.clone()
```
and make it work by supporting tests on uncoalesced tensors (use `_values()` and `_indices()` to get the member data of uncoalesced tensors)

Reply:
`actual` is not coalesced here:

```python
import torch

indices = (
    (0, 1, 0,),
    (1, 0, 1,),
)
values = (1, 1, 1)
actual = torch.sparse_coo_tensor(indices, values, (2, 2))

assert not actual.is_coalesced()
```

But I agree, we need to get the general logic sorted out first and than have another go at this test.

Reply:
> `actual` is not coalesced here:

Sure, not here, but `actual` is currently always coalesced when performing the `assert_close` check. So, effectively, it is coalesced, IMHO.

Reply:
The purpose of this test was to check if sparse tensors are coalesced before their members are being compared. If that would not be the case the test will fail.